GVFS/GVFS.Common/LibGit2RepoPool.cs

```diff
@@ -9,32 +9,46 @@ namespace GVFS.Common
     public class LibGit2RepoPool
     {
         private const int TryAddTimeoutMilliseconds = 10;
-        private const int TryTakeTimeoutMilliseconds = Timeout.Infinite;
-        
+        private const int TryTakeWaitForever = Timeout.Infinite;
+        private const int TryTakeWaitShort = 10;
+
         private readonly BlockingCollection<LibGit2Repo> pool;
+        private readonly Func<LibGit2Repo> repoFactory;
         private readonly ITracer tracer;
 
+        private readonly Timer repoDisposalTimer;
+        private readonly TimeSpan repoDisposalDueTime = TimeSpan.FromSeconds(15);

```

Comment:
(My latest push comes up with potentially good numbers, but this should be a big focus for discussion.)

Reply:
@wilbaker (You may have missed this comment because it is out-dated... I don't actually want these values to be so short. Instead, we should try to find out good values that are reasonable.)